pytube/streams.py

```diff
@@ -334,8 +341,10 @@ def download(
                     # send to the on_progress callback.
                     self.on_progress(chunk, fh, bytes_remaining)
         self.on_complete(file_path)
+        self.stop_signal = False        
         return file_path
-
+    def stop_download(self):
+        self.stop_signal = True
```

Comment:
Formatting (new lines) and what calls this? Is there signal handling?

Reply:
The call is made using a Stream object by the user, What do you mean by signal handling?